src/OpenTelemetry.Contrib.Extensions.AWSXRay/Resources/AWSECSResourceDetector.cs

```diff
@@ -34,18 +38,30 @@ public class AWSECSResourceDetector : IResourceDetector
     /// <returns>List of key-value pairs of resource attributes.</returns>
     public IEnumerable<KeyValuePair<string, object>> Detect()
     {
-        List<KeyValuePair<string, object>> resourceAttributes = null;
-
         if (!this.IsECSProcess())
         {
-            return resourceAttributes;
+            return new List<KeyValuePair<string, object>>();
```

Comment:
Changed behavior from returning null to empty list. Is it expected?

Reply:
It is fine according to the API, and null-safe is usually better that nullable, but it is not a necessary change by any means

Reply:
I reverted the change in behavior and align it with the other detectors. I like more detectors returning an empty resource than a `null` one for nullability, but is not something I would cling to in a PR like this.